doc/console/serial-console-HLD.md

```diff
@@ -0,0 +1,214 @@
+| Policy             | Action                                                                | Param values        | Default      |
+|--------------------|-----------------------------------------------------------------------|---------------------|--------------|
+| auto logout        | Inactivity timeout for serial-console session                         | 0-35000 (min)       | 15           |
+| sysrq capabilities | Enabling or disabling SysRq functionality for serial-consoles         | enabled/disabled    | disabled     |
+
+
+##  4. <a name='Init flow'></a>Init flow 
+
+During init flow we will set default configurations as mentioned in the table abowe. Default values will be added to init_cfg.json.j2 .
+###  4.1. <a name='FeatureDefault'></a>Feature Default
+
+Description of default values in init_cfg.json regarding serial-console config:
+```
+auto logout: 15 //minutes
+sysrq capabilities: disabled
+```
+##  5. <a name='SAI api'></a>SAI api
+NA
+##  6. <a name='Configurationandmanagement'></a>Configuration and management 
+
+###  6.1. <a name='SERIAL_CONSOLEconfigDBtable'></a>SERIAL_CONSOLE configDB table
+
+```
+SERIAL_CONSOLE:{
+	"auto_logout": {{min}}
+	"sysrq-capabilities": {{enabled|disabled}}
+}
+```
+### 6.2. ConfigDB schemas
+```
+; Defines schema for SERIAL_CONSOLE configuration attributes in SERIAL_CONSOLE table:
+auto_logout                           = 5*DIGIT                 ; autologout timer - max is 35000 minutes
+max_syslogins                         = "enabled/disabled"      ; enabling or disabling sysrq capabilities in linux, default is disabled.  
+```
+
+###  6.3. <a name='CLIYANGmodelEnhancements'></a>CLI/YANG model Enhancements
+```yang
+//filename:  sonic-serial-console.yang
+module sonic-serial-console {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-serial-console";
+    prefix cli-sessions;
+
+    description "SERIAL_CONSOLE YANG Module for SONiC-based OS";
+
+    revision 2023-06-07 {
+        description "First Revision";
+    }
+
+    container sonic-serial-console {
+
+        container SERIAL_CONSOLE {
+
+            description "SERIAL_CONSOLE part of config_db.json";
+
+            key "name";
+
+            leaf auto_logout {
+                description "serial-console auto-logout timer value in minutes";
+                type int32 {
+                    range "0..15000";
+                }
+                default 15;
+            }
+
+            leaf sysrq_capabilities {
+                description "managing SysRq capabilities";
+                type enumeration {
+                    enum disabled;
+                    enum enabled;
+                }
+                default disabled;
+            }
+        }
+        /* end of container SERIAL_CONSOLE */
+    }
+    /* end of top level container */
+}
+/* end of module sonic-serial-console */
+
+```
+###  6.4. <a name='ConfigDBEnhancements'></a>Config DB Enhancements
```

Comment:
can we use the native markdown format instead of http marker? there are other cases, please correct as well.

Reply:
Done.